bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+- otherwise:
+    - if `announcement_sig` is invalid OR NOT correct:
+        - SHOULD send a `warning`.
+        - MAY close the connection.
+        - MUST ignore the message.
+    - otherwise:
+        - if `node_id_1` OR `node_id_2` are blacklisted:
+            - SHOULD ignore the message.
+        - otherwise:
+            - if the transaction referred to was NOT previously announced as a
+              channel:
+                - SHOULD queue the message for rebroadcasting.
+                - MAY choose NOT to for messages longer than the minimum
+                  expected length.
+        - if it has previously received a valid `channel_announcement_2`, for
+          the same transaction, in the same block, but for a
+          different `node_id_1` or `node_id_2`:
+            - SHOULD blacklist the previous message's `node_id_1`
+              and `node_id_2`, as well as this `node_id_1` and `node_id_2` AND
+              forget any channels connected to them.
+        - otherwise:
+            - SHOULD store this `channel_announcement_2`.
+- once its funding output has been spent OR reorganized out:
+    - SHOULD forget a channel after a 12-block delay.
+
+### Rationale
+
+Both nodes are required to sign to indicate they are willing to route other
+payments via this channel (i.e. be part of the public network); requiring their
+aggregated MuSig2 Schnorr signature proves that they control the channel.
+
+The blacklisting of conflicting nodes disallows multiple different
+announcements. Such conflicting announcements should never be broadcast by any
+node, as this implies that keys have leaked.
+
+While channels should not be advertised before they are sufficiently deep, the
+requirement against rebroadcasting only applies if the transaction has not moved
+to a different block.
+
+In order to avoid storing excessively large messages, yet still allow for
+reasonable future expansion, nodes are permitted to restrict rebroadcasting
+(perhaps statistically).
+
+New channel features are possible in the future: backwards compatible (or
+optional) features will have _odd_ feature bits, while incompatible features
+will have _even_ feature bits
+(["It's OK to be odd!"](00-introduction.md#glossary-and-terminology-guide)).
+
+A delay of 12-blocks is used when forgetting a channel on funding output spend
+as to permit a new `channel_announcement_2` to propagate which indicates this
+channel was spliced.
+
+## The `node_announcement_2` Message
+
+TODO: complete
+
+1. type: xxx (`node_announcement_2`)
+2. data: (tlv_stream):
+    1. type: 0 (`announcement_sig`)
+    2. data:
+        * [`schnorr_signature`:`schnorr_signature`]
+    3. type: 1 (`features`)
+    4. data:
+        * [`...*byte`: `features`]
+    5. type: 2 (`timestamp`)
+    6. data:
+        * [`u32`: `block_height`]
+    7. type: 4 (`color`)
+    8. data:
+        * [`3*byte`:`rgb_color`]
+    9. type: 5 (`alias`)
+    10. data:
+         * [`32*byte`:`alias`]
```

Comment:
Make this ...*byte?  And then restrict it to <= 32 bytes?  That way we don't have to pad?

Reply:
See also https://github.com/lightning/bolts/pull/798/commits/d2ba8ebe0fe4b512114222b8b3cc88c227d7c001 which defines a "utf8" type which would be great here.